src/Magento/FunctionalTestingFramework/Util/DocGenerator.php

```diff
@@ -0,0 +1,193 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\FunctionalTestingFramework\Util;
+
+use Magento\FunctionalTestingFramework\Exceptions\TestFrameworkException;
+use Magento\FunctionalTestingFramework\Test\Handlers\ActionGroupObjectHandler;
+use Magento\FunctionalTestingFramework\Test\Objects\ActionGroupObject;
+use Magento\FunctionalTestingFramework\Test\Objects\TestObject;
+
+/**
+ * Class TestGenerator
+ * @SuppressWarnings(PHPMD)
+ */
+class DocGenerator
+{
+    const DEFAULT_OUTPUT_DIR =
+        PROJECT_ROOT .
+        DIRECTORY_SEPARATOR .
+        "dev" .
+        DIRECTORY_SEPARATOR .
+        "tests" .
+        DIRECTORY_SEPARATOR .
+        "docs";
+    const DOC_NAME = "documentation.md";
+    const ANNOTATION_MODULE = "";
+    const ANNOTATION_PAGE = "page";
+    const ANNOTATION_DESCRIPTION = "description";
+    const FILENAMES = "filenames";
+    const ARGUMENTS = "arguments";
+
+    /**
+     * Single instance of class var
+     *
+     * @var ActionGroupObjectHandler
+     */
+    private static $instance;
```

Comment:
Does this need to be a singleton as opposed to a static class that you could call `DocGenerator::generateDocs()`? The only public function you have in this class is passed in all it needs to generate the docs, and as far as I can tell this class does not store any data for long term use (which imo defeats the purpose of the singleton).